packages/cloud-functions/src/firebase.ts

```diff
@@ -0,0 +1,19 @@
+import * as admin from 'firebase-admin'
+import * as functions from 'firebase-functions'
+
+admin.initializeApp(functions.config().firebase)
+
+export function saveTxHashProvider(address: string, txHash: string, provider: string) {
+  admin
+    .database()
+    .ref(`/registrations/${address}/txHashes/${txHash}`)
```

Comment:
Out of curiosity - is there any added value in storing the provider `txHash` nested under an individual `address`? I'm probably missing something but wouldn't it be more simple to just have a simple and independent table with columns:`txHash`, `timestamp`,  and `provider`?

Reply:
With this approach we make a single query to get the whole `txHashes` for the address and keep that in the redux store. If we kept it on a separate table we'd have to either get the whole table which could become very large with all CICO transactions over time or make a query for every tx hash.

The timestamp matching can be done in the webhook itself to find the tx hash of the purchase and put it here as well so that we don't have that logic in the client.

Reply:
Got it - makes sense this organization is more efficient from a lookup perspective.

Will matching the timestamp in the webhook itself require us to tap into Blockscout data? Seems like this will only be an issue for Simplex so not really relevant for this PR but just curious

Reply:
Yes, we'll have to query the blockchain and for that we can use either Blockscout or Forno.